raft/server.hh

```diff
@@ -43,8 +55,15 @@ public:
         bool enable_forwarding = true;
 
         // Max size of a single command, add_entry with a bigger command will throw command_is_too_big_error.
-        // The value of zero means no limit.
-        size_t max_command_size = 0;
+        // The following condition must be satisfied:
+        // max_command_memory_usage() <= max_log_size - snapshot_trailing_size
```

Comment:
It's sizeof is not, or at least should not be. But also how a user of the library suppose to know that he has to take into account log_entry size? Look at the source? I know you like to do it, but the interface should make sense without.

Reply:
> But also how a user of the library suppose to know that he has to take into account log_entry size?

He'll read the comments in `server::configuration`? It's explicitly stated there.

`I know you like to do it, but the interface should make sense without.`

I don't like doing it, and I don't find it pretty either, but I don't see how to do it better yet, your advice is welcome.

Reply:
I suggested previous to simply not account for metadata at all. There are other metadata associated with storing an entry (pointer size, container internal memory) that we do not account for. The memory usage is not exact. It means to provide a limit, but not a hard limit. If your concern is that we may add a lot of 1 bytes commands and the overhead will be a substantial part of the memory usage (which is also be true for metadata we do not take into account btw) we may apply the overhead calculation selectively. If a command it smaller than `max_command_size - sizeof(log_entry)` take sizeof(log_entry) into account, otherwise do not. Can this work?

Reply:
> Can this work?

It seems it can, I'll try it.

Reply:
done